test/docker-abstraction-guard.test.ts

```
@@ -0,0 +1,65 @@
+    );
+
+    if (directDockerCalls.length > 0) {
+      const formatted = directDockerCalls
+        .map(
+          (entry) =>
+            `  - ${entry.filePath}:${String(entry.line)}:${String(entry.column)} ${entry.expression}: ${entry.snippet}`,
+        )
+        .join("\n");
+      throw new Error(
+        `Direct docker process launches must use the src/lib/docker abstractions.\n` +
+          `Do not add calls like run([\"docker\", ...]), runCapture([\"docker\", ...]), ` +
```

ESLint

[error] Unnecessary escape character: \".

(no-useless-escape)